var/spack/repos/builtin/packages/esmf/package.py

```diff
@@ -15,60 +15,67 @@ class Esmf(MakefilePackage):
     software infrastructure for building and coupling weather, climate, and
     related Earth science applications. The ESMF defines an architecture for
     composing complex, coupled modeling systems and includes data structures
-    and utilities for developing individual models."""
+    and utilities for developing individual models.
+    The National Unified Operational Prediction Capability (NUOPC) Layer
+    defines a common model architecture to support interoperable ESMF components.
+    The NUOPC Layer is included with the ESMF package."""
 
-    homepage = "https://www.earthsystemcog.org/projects/esmf/"
+    homepage = "https://earthsystemmodeling.org/"
     url = "https://github.com/esmf-org/esmf/archive/ESMF_8_0_1.tar.gz"
     git = "https://github.com/esmf-org/esmf.git"
 
-    maintainers("climbfuji", "jedwards4b", "AlexanderRichert-NOAA")
+    maintainers("climbfuji", "jedwards4b", "AlexanderRichert-NOAA", "theurich", "uturuncoglu")
 
     # Develop is a special name for spack and is always considered the newest version
     version("develop", branch="develop")
     # generate chksum with spack checksum esmf@x.y.z
     version("8.4.1", sha256="1b54cee91aacaa9df400bd284614cbb0257e175f6f3ec9977a2d991ed8aa1af6")
     version("8.4.0", sha256="28531810bf1ae78646cda6494a53d455d194400f19dccd13d6361871de42ed0f")
     version("8.3.1", sha256="6c39261e55dcdf9781cdfa344417b9606f7f961889d5ec626150f992f04f146d")
-    version("8.3.0", sha256="0ff43ede83d1ac6beabd3d5e2a646f7574174b28a48d1b9f2c318a054ba268fd")
-    version("8.3.0b09", commit="5b7e546c4b")
+    version(
+        "8.3.0",
+        sha256="0ff43ede83d1ac6beabd3d5e2a646f7574174b28a48d1b9f2c318a054ba268fd",
+        deprecated=True,
+    )
+    version("8.3.0b09", commit="5b7e546c4b", deprecated=True)
```

Comment:
Note that his is still the version used by the ufs-weather-model develop branch as of today. We shouldn't deprecate it until the UFS WM moved away (because we need to install it everywhere as part of our spack-stack releases, and I don't want to add the flag to install deprecate packages as a blanket for the entire environment).

Reply:
@climbfuji I see. I had thought that ufs-weather-model develop branch had moved on to ESMF 8.4.1 now -  @junwang-noaa can you help clarify?
My rationale for deprecation marking in this PR was that if there was a patch release, e.g. 8.3.1, then mark 8.3.0 deprecated (same for 8.4.1 over 8.4.0) simply because if there is a patch, it meant that there was actually an issue found in the .0 release that needed fixing. Therefore marked deprecated.
But I also understand your concern with regards to downstream changes needed if ufs-weather-model is right at 8.3.0.
Let's clarify this, and if the impact is unreasonable, then remove the `deprecated=True` from the version that you need for continued support.

Reply:
@theurich  We have a ufs-weather-model branch with ESMF 8.4.1 tested. We are waiting for the libraries to be installed on wcoss2 before we can commit the code.

Reply:
Thanks @junwang-noaa. So @climbfuji, it looks like this might come down to a timing question. Jun, how close do you think we are with v8.4.1 as the official ESMF dependency for ufs-weather-model? A matter of days, or more like weeks?
Also, just to clarify, it is actually v8.3.0b09 that is currently the official dependency, right? Definitely be great to get away from a beta tag, and mark it as deprecated when we can... but I am okay to have it as a follow-on PR if timing is better that way. Better than holding up this PR.

Reply:
It’s a bit of a chicken-and-egg problem. esmf 8.4.1 will be made available with spack-stack, but in order to switch to spack-stack it is easier to have 8.3.0b09. However, if you make the change now it won’t go into our release branch (but come to develop later), so maybe after all just leave it like it is.

And maybe deprecating that version will speed up the update in the ufs-weather-model!